decredplugin/decredplugin.go

```diff
@@ -273,14 +317,108 @@ func (s *StartVote) VerifySignature() error {
 	return nil
 }
 
-// EncodeStartVoteencodes StartVoteinto a JSON byte slice.
-func EncodeStartVote(v StartVote) ([]byte, error) {
+// EncodeStartVoteV1 encodes a StartVoteV1 into a JSON byte slice.
+func EncodeStartVoteV1(v StartVoteV1) ([]byte, error) {
+	return json.Marshal(v)
+}
+
+// DecodeVotedecodes a JSON byte slice into a StartVoteV1.
+func DecodeStartVoteV1(payload []byte) (*StartVoteV1, error) {
+	var sv StartVoteV1
+
+	err := json.Unmarshal(payload, &sv)
+	if err != nil {
+		return nil, err
+	}
+
+	return &sv, nil
+}
+
+// VoteV2 represents the vote options and vote parameters for a StartVoteV2.
+//
+// Differences between VoteV1 and VoteV2:
+// * Added the ProposalVersion field that specifies the version of the proposal
+//   that is being voted on. This was added so that the proposal version is
+//   explicitly included in the StartVote signature.
+// * Added a Type field in order to specify the vote type.
+type VoteV2 struct {
```

Comment:
Why is there no version in this struct?

Reply:
The Vote struct is part of the StartVote struct which contains the version.

```
type StartVoteV2 struct {
	// decred plugin only data
	Version uint `json:"version"` // Version of this structure

	PublicKey string `json:"publickey"` // Key used for signature
	Vote      VoteV2 `json:"vote"`      // Vote options and params
	Signature string `json:"signature"` // Signature of Vote hash
}
```

Do we need to version the Vote struct as well?

Reply:
Got it.